src/ol/layer/group.js

```diff
@@ -221,6 +221,10 @@ ol.layer.Group.prototype.getLayerStatesArray = function(opt_states) {
         layerState.extent = ownLayerState.extent;
       }
     }
+    if (layerState.layer.getZIndex() === undefined) {
+      layerState.zIndex = ownLayerState.zIndex;
+      layerState.layer.setZIndex(ownLayerState.zIndex);
```

Comment:
I addressed this in [Commit-b843d7d](https://github.com/openlayers/openlayers/pull/6661/commits/b843d7de52e4ea89746c05dbdef2b8f31ea57293)